src/plugins/intel_gna/backend/make_pwl.cpp

```diff
@@ -266,20 +266,36 @@ void make_gna_pwl(const DnnActivation&  fun,
             gna_pwl[0].slope = 0;
             print_segment(gna_pwl[0].xBase / in_scale, gna_pwl[0].yBase / out_scale, 0.0);
 
-            gna_pwl[1].xBase = x_lower & XBASEMASK;  // zero out the 2 lsb
-            gna_pwl[1].yBase = y_lower;
+            int32_t xBase = x_lower & XBASEMASK;  // zero out the 2 lsb
+            int16_t yBase = y_lower;
             s = gna_slope(1.0, in_scale, out_scale);
-            gna_pwl[1].slope = FLOAT_TO_INT16(s.slope * s.slope_scale);
-            gna_pwl[1].xBase = gna_pwl[1].xBase | s.slope_scale_index;
-            print_segment((int32_t)(gna_pwl[1].xBase & XBASEMASK) / in_scale, gna_pwl[1].yBase / out_scale, 1.0);
+            int16_t slope = FLOAT_TO_INT16(s.slope * s.slope_scale);
+
+            gna_pwl[1].xBase = xBase | s.slope_scale_index;
+            gna_pwl[1].yBase = yBase;
+            gna_pwl[1].slope = slope;
+            print_segment(static_cast<int32_t>(gna_pwl[1].xBase & XBASEMASK) / in_scale,
+                          gna_pwl[1].yBase / out_scale,
+                          1.0);
+
+            // It was decided based on 89164 to introduce extra segment to ensure that PWL will cross (0,0).
+            // count zero of function for middle segment.
+            if (fun == kActIdentity) {
+                auto x0 = -yBase / s.slope + xBase;  // s.slope is double so whole statement is double
+                if (x0 != 0.0) {
+                    // xBase is 0 so it does not make sense to apply XBASEMASK
+                    // reuse the same slope as for previous segment
+                    gna_pwl.push_back({static_cast<int32_t>(0 | s.slope_scale_index), 0, slope});
```

Comment:
@mdeisher AFAIR this is partially outdated since we fixed it already with @marcinkusm.

Beside, how can split (at INT32_MIN/2) guarantee to pass through (0,0)? IMO even it we do it iteratively, such algorithm is complex and has potential effect of not ensuring to pass through (0,0).

Reply:
@mdeisher could you help me and suggest some values/parameters which could provide results you mentioned could give positive value for f(-1) it would help me to verify my solution and write unit tests.

Reply:
@marcinkusm, any roundoff error causing an increase in slope would cause this.  Testing for it is trivial.  If f(-1)>f(0) then the function is non-monotonic.  Keep in mind that when I say -1 and 0, I mean integer -1 and integer 0.  Please let's discuss if it is not perfectly clear.

Reply:
@kbruniec, please explain how my comment is outdated.  I outlined the way to guarantee that the segment passes through integer (0,0).  It can be done by adjusting the yBase.  The problem with adjusting yBase to fix the right end is that it introduces error at the left end.  That is why I suggested splitting the segment if the error at the left end is too large.  The algorithm is not complex at all.

Reply:
@mdeisher I didn't said your comment is outdated. I meant the new code was partially outdated (not pushed).

Reply:
Implemented new algorithm which adapt x_base on the left and right side of (0,0) in case of insertion of (0,0)

Reply:
implementation done as agreed, continuation of discussion about algorithm is continued in following thread

https://github.com/openvinotoolkit/openvino/pull/12825#pullrequestreview-1121946458